spring-webmvc/src/main/java/org/springframework/web/servlet/DispatcherServlet.java

```diff
@@ -1046,10 +1047,9 @@ protected void doDispatch(HttpServletRequest request, HttpServletResponse respon
 
 				// Process last-modified header, if supported by the handler.
 				String method = request.getMethod();
-				boolean isGet = "GET".equals(method);
-				if (isGet || "HEAD".equals(method)) {
+				if (HttpMethod.GET.matches(method) || HttpMethod.HEAD.matches(method)) {
 					long lastModified = ha.getLastModified(request, mappedHandler.getHandler());
-					if (new ServletWebRequest(request, response).checkNotModified(lastModified) && isGet) {
+					if (new ServletWebRequest(request, response).checkNotModified(lastModified) && HttpMethod.GET.matches(method)) {
```

Comment:
Can you please update this to avoid matching GET twice by using a local variable as in the current code?

Reply:
Thanks a lot for the review. I applied it.